lib/Command/RepairNotifications.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * @author Jannik Stehle <jstehle@owncloud.com>
+ * @author Jan Ackermann <jackermann@owncloud.com>
+ *
+ * @copyright Copyright (c) 2021, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Notifications\Command;
+
+use OCP\IDBConnection;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class RepairNotifications extends Command {
+
+	/** @var IDBConnection */
+	protected $connection;
+
+	public static $availableSubjects = [
+		'relativeLinks'
+	];
+
+	/**
+	 * @param IDBConnection $connection
+	 */
+	public function __construct(IDBConnection $connection) {
+		parent::__construct();
+		$this->connection = $connection;
+	}
+
+	protected function configure() {
+		$this
+			->setName('notifications:repairNotifications')
+			->setDescription('Repair existing notifications')
+			->addArgument('subject', InputArgument::REQUIRED, 'Subject to repair')
+		;
+	}
+
+	/**
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return false|int
+	 */
+	protected function execute(InputInterface $input, OutputInterface $output) {
+		$subject = $input->getArgument('subject');
+
+		if (!\in_array($subject, self::$availableSubjects)) {
+			throw new \LogicException('Invalid subject');
```

Comment:
I think it's preferred to write an error message and return a non-zero value, specially in this case.
Throwing an exception here doesn't add any additional value. In addition, unhandled exceptions like this one usually look bad because they aren't handled.